src/python/turicreate/_deps/__init__.py

```diff
@@ -275,7 +275,12 @@ def __get_version(version):
 
 def __has_module(name):
     if six.PY2:
-        return imp.find_module(name) is not None
+        try:
+            spec = imp.find_module(name)
+            # file or module path
```

Comment:
I don't understand this comment or the conditional below it.

Reply:
from [imp.find_module](https://docs.python.org/2/library/imp.html) doc,
```
If search is successful, the return value is a 3-element tuple (file, pathname, description):

file is an open file object positioned at the beginning, pathname is the pathname of the file found, and description is a 3-element tuple as contained in the list returned by get_suffixes() describing the kind of module found.

If the module does not live in a file, the returned file is None, pathname is the empty string, and the description tuple contains empty strings for its suffix and mode; the module type is indicated as given in parentheses above. If the search is unsuccessful, ImportError is raised. Other exceptions indicate problems with the arguments or environment.

If the module is a package, file is None, pathname is the package path and the last item in the description tuple is PKG_DIRECTORY.
```

Reply:
spec[0] is file and spec[1] is package (or module) path.

Reply:
Perhaps the following would be clearer:
```python
(file, pathname, _) = imp.find_module(name)
return not (file is None and pathname is None)
```

Reply:
Perfect! I will use this form. Thanks!